### PR TITLE
Fix overlapping styles/entities

### PR DESCRIPTION
**Pull request by ryanscottaudio**
Fix calculation of prefix/suffix length for entity components. Addresses https://github.com/HubSpot/draft-convert/issues/76.

### CODE REVIEW

**Comment by benbriggs** [comment]
Good catch! Sorry for the delay 😞 

After doing some investigating, this looks like approach. My one question is around the recursive call of `getElementTagLength` - is that solving an instance where you wrap an entity in 2+ layers of tags? My first impression is that it shouldn't count any other layers of components since the only ones that can be present are from that single mutation.

**Comment by ryanscottaudio** [comment]
@benbriggs we have instances in our use case where an element may have multiple layers, so we need to make sure we're getting down to the raw text of the children

**Comment by ryanscottaudio** [comment]
@benbriggs there was actually more work we did on this having to do w/ special characters, let me merge that stuff too

File: test/spec/convertToHTML.js [hunk 1]
```diff
@@ -177,118 +177,6 @@ describe('convertToHTML', () => {
     expect(result).toBe('<ul><li>top level</li><ul><li>nested one level</li><ul><li>nested two levels</li></ul></ul><li>back to top level</li></ul>');
   });
 
-  it('escapes HTML in text of blocks', () => {
```
Comment (benbriggs) [inline test/spec/convertToHTML.js]: hmm, what's the context on removing these? this is a pretty important XSS protection

**Comment by benbriggs** [comment]
Just opened #88 because theres some other stuff in here that would block the original fix's release. I'd be interested in hearing the context behind any other changes before merging them. Thanks for digging into the issue!

**Pull request closed without merging**